src/core/featurehistory.h

```diff
@@ -0,0 +1,124 @@
+#ifndef FEATUREHISTORY_H
+#define FEATUREHISTORY_H
+
+#include <QObject>
+#include <QTimer>
+#include <qgsproject.h>
+
+
+typedef QPair<QgsFeature, QgsFeature> OldNewFeaturePair;
+
+class FeatureHistory : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY( bool isUndoEnabled READ isUndoEnabled NOTIFY isUndoEnabledChanged )
```

Comment:
I think I'd rename this to is{Undo, Redo}Available

Reply:
Renamed.